### ISSUE TITLE
Does the script file contail FT step?

### ISSUE THREAD

**Issue by wjczf123** [AtmaHou/FewShotJoint#4]
Hi, yutai. I'm very sorry to interrupt again.

I reproduced your code on the SNIPS dataset under 5-shot setting (snips_ConProm+TR_5.sh). But the result looks like ConProm+TR instead of ConProm+FT+TR. The three results are: 
1. test:{'slot': 0.6433199999999999, 'slot_p': 62.451, 'slot_r': 66.438, 'intent': 0.986999999026, 'sentence_acc': 0.396} 
2. test:{'slot': 0.6876900000000001, 'slot_p': 65.84100000000001, 'slot_r': 72.00200000000001, 'intent': 0.9889999990220002, 'sentence_acc': 0.468}
3. test:{'slot': 0.6587799999999999, 'slot_p': 60.845000000000006, 'slot_r': 71.896, 'intent': 0.9909999990180001, 'sentence_acc': 0.395}
Does this script file not contain FT step? 

Thanks.

**Comment by AtmaHou** [comment]
Yep, as mentioned in the [Readme's results](https://github.com/AtmaHou/FewShotJoint#fewjoint11), the code only contains ConProm+TR for code-simplicity. 
And you can implement +FT by fine-tuning the model on the test domain support features with the trainer module.

**Comment by wjczf123** [comment]
OK. Thank you a lot.

**Comment by AtmaHou** [comment]
> Yep, as mentioned in the [Readme's results](https://github.com/AtmaHou/FewShotJoint#fewjoint11), the code only contains ConProm+TR for code-simplicity. And you can implement +FT by fine-tuning the model on the test domain support features with the trainer module.

Note: As mentioned in the paper, for better results and to avoid overfitting, we are using a MAML-like learning process for finetune. The finetuning process we ended up using was:

1. learn the encoder initialization parameters in the source domain using MAML
2. learning in the source domain using ConProm's learning target
3. finetune the encoder with MAML on ONE support set using the SLU task (normal intent classification and slot filling)
4. test on the corresponding query set

This process finetunes the encoder only. The upper-structures are not finetuned because it is not convenient to finetune the metric learning targets only with support set. Also, these upper-structures are meant to learn the domain-general ability to find correlations based on input similarity, and may be less likely to overfit without finetuning.

**Comment by wjczf123** [comment]
OK. Thanks.
